bucket/covergroup.py

```diff
@@ -192,7 +182,24 @@ def exclude_by_tags(self, tags: TagStrs, match_all: bool = False):
             tags: Tag(s) to match against
             match_all: If set, all tags must match. If cleared, any tags can match
         """
-        return self.exclude_by_function(self._match_by_tags(tags))
+        return self.exclude_by_function(self._match_by_tags(tags, match_all))
+
+    @validate_call
+    def set_tier_level(self, tier: int):
+        """
+        Filter the coverage tree for coverpoints equal to or less than 'tier'.
+        Parameters:
+            tier: The highest tier level to be set active
+        """
+        self._set_tier_level(tier)
+        return self
+
+    def _set_tier_level(self, tier: int):
+        any_children_active = False
+        for child in self.iter_children():
+            any_children_active |= child._set_tier_level(tier)
+        self.tier_active = any_children_active
+        return self.tier_active
```

Comment:
I think this is the right approach, but I don't think there's any point having a separate `_set_tier_level` - just put the functionality in `set_tier_level`.

Worth also adding the interface to CoverBase if it's meant to be on both

Reply:
I split it out into user-facing function and internal function so the internal could be easily changed without needing the user to update their end. 
Also set_tier_level returns self, _set_by_tier returns bool to pass upwards.

Reply:
👍 didn't notice the return